libraries/botframework-connector/src/auth/userTokenClientImpl.ts

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { Activity, SignInUrlResponse, TokenExchangeRequest, TokenResponse, TokenStatus } from 'botframework-schema';
+import { ServiceClientCredentials } from '@azure/ms-rest-js';
+import { TokenApiClient } from '../tokenApi/tokenApiClient';
+import { UserTokenClient } from './userTokenClient';
+import { assert } from 'botbuilder-stdlib';
+
+// Internal
+export class UserTokenClientImpl extends UserTokenClient {
+    private readonly client: TokenApiClient;
+    constructor(private readonly appId: string, credentials: ServiceClientCredentials, oauthEndpoint: string) {
+        super();
+        this.client = new TokenApiClient(credentials, { baseUri: oauthEndpoint });
+    }
+
+    async getUserToken(
+        userId: string,
+        connectionName: string,
+        channelId: string,
+        magicCode: string
+    ): Promise<TokenResponse> {
+        assert.string(userId, ['userId']);
+        assert.string(connectionName, ['connectionName']);
+        assert.string(channelId, ['channelId']);
+
+        const result = await this.client.userToken.getToken(userId, connectionName, { channelId, code: magicCode });
+        return result._response.parsedBody;
+    }
+
+    async getSignInResource(
+        connectionName: string,
+        activity: Activity,
+        finalRedirect: string
+    ): Promise<SignInUrlResponse> {
+        assert.string(connectionName, ['connectionName']);
+        assert.object(activity, ['activity']);
+
+        const result = await this.client.botSignIn.getSignInResource(
+            UserTokenClient.createTokenExchangeState(this.appId, connectionName, activity),
+            { finalRedirect }
+        );
+
+        return result._response.parsedBody;
+    }
+
+    async signOutUser(userId: string, connectionName: string, channelId: string): Promise<void> {
+        assert.string(userId, ['userId']);
+        assert.string(connectionName, ['connectionName']);
+        assert.string(channelId, ['channelId']);
+
+        const result = await this.client.userToken.signOut(userId, { channelId, connectionName });
+        return result._response.parsedBody;
```

Comment:
the return type is `Promise<void>` - is this intentional?

Reply:
Nope, thanks for catching it!